### PR TITLE
chore(job): allow server-side validation of AWS-flavored cron

### PR DESCRIPTION
**Pull request by bvtujo**
Goal: Allow customers to specify any AWS-flavored cron they like, even though it contains nonstandard wildcards and an extra "year" field. Currently, we can't support the `L` and `W` wildcards, or jobs which run only in certain years. 

This PR addresses this lack by allowing customers to wrap a schedule expression in `rate()` or `cron()` decoration to bypass client-side validation (which uses a standard Unix cron parser) and let CloudWatch Events validate the cron expression. This extends the time to deployment failure for this special case by approximately 30 seconds. 

Customers can now specify this either in flags:
```
copilot job init -n mailer --schedule "cron(0 12 L * ? 2021)" // run this at noon on the last day of every month in 2021
```
Or by changing the schedule in the manifest:
```yaml
schedule: rate(3 hours) // Run this every three hours; equivalent to "@every 3h"
```

#1131  related #1451 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/validate.go [hunk 1]
```diff
@@ -231,6 +233,13 @@ func basicNameValidation(val interface{}) error {
 }
 
 func validateCron(sched string) error {
+	// If the schedule is wrapped in aws terms `rate()` or `cron()`, don't validate it--
+	// instead, pass it in as-is for serverside validation. AWS cron is weird (year field, nonstandard wildcards)
+	// so for edge cases we need to support it
+	awsSchedMatch := awsScheduleRegexp.FindStringSubmatch(sched)
```
Comment (efekarakus) [inline internal/pkg/cli/validate.go:239]: Maybe we can create a `manifest/schedule` package with a `ValidateExpr(expr string) error` function, that has the common functionality between `cli/validate.go` and `stack/scheduled_job.go`.

Probably worthwhile if you think there are other neat things we can put in the `schedule` package

**Review by efekarakus — APPROVED** [review]

**Comment by huanjani** [comment]
This is cool!
Do we need to change messaging to users so they know this flavor is available?

File: internal/pkg/cli/validate.go (hunk 1 above)
Comment (bvtujo) [reply internal/pkg/cli/validate.go:239]: I think for the time being I'll leave it as is, but I could see for instance an `manifest/events` package for jobs which handles the logic for parsing a yaml description of a CW Event Rule based on arbitrary event patterns, which could also include schedule validation.

**Comment by bvtujo** [comment]
@huanjani yeah, good callout. Updated the manifest comment and flag description.

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/cli/flag.go [hunk 2]
```diff
@@ -217,7 +217,9 @@ Tasks with the same group name share the same set of resources.
 Accepts valid Go duration strings. For example: "2h", "1h30m", "900s".`
 	scheduleFlagDescription = `The schedule on which to run this job. 
 Accepts cron expressions of the format (M H DoM M DoW) and schedule definition strings. 
-For example: "0 * * * *", "@daily", "@weekly", "@every 1h30m".`
+For example: "0 * * * *", "@daily", "@weekly", "@every 1h30m".
+AWS Schedule Expressions of the form "rate(10 minutes)" or "cron(0 12 L * ? 2021)"
+are also accepted.`
```
Comment (huanjani) [inline internal/pkg/cli/flag.go:222]: nit: Should this more closely parallel the wording/capitalization in the preceding lines? ("AWS schedule expressions are also accepted. For example: "rate(10 minutes)" or "cron(0 12 L * ? 2021)"" or something like that?)

**Pull request merged**